File: src/main/java/com/qtech/bubbles/common/filefilters/VisibilityFileFilter.java
package com.qtech.bubbles.common.filefilters;

import java.io.File;
import java.io.FileFilter;

public record VisibilityFileFilter(boolean filterVisible) implements FileFilter {
    @Override
    public boolean accept(File pathname) {
        return filterVisible != pathname.isHidden();
    }
}

File: src/main/kotlin/qtech/bubbles/common/filefilters/VisibilityFileFilter.kt
package qtech.bubbles.common.filefilters

import java.io.File
import java.io.FileFilter

class VisibilityFileFilter(private val filterVisible: Boolean) : FileFilter {
    override fun accept(pathname: File): Boolean {
        return if (filterVisible) {
            !pathname.isHidden
        } else {
            pathname.isHidden
        }
    }
}